src/loading.jl

```diff
@@ -0,0 +1,47 @@
+_loadleaf(x) = isleaf(x)
+for T in [:Dense, :Diagonal, :Bilinear, :Embedding,
+          :Conv, :ConvTranspose, :DepthwiseConv, :CrossCor]
+  @eval _loadleaf(::$T) = true
+end
+
+loadto!(x, x̄) = x
+loadto!(x::Zeros, x̄) = x
+loadto!(x::AbstractArray, x̄::AbstractArray) = copyto!(x, x̄)
+for T in [:Dense, :Bilinear, :Conv, :ConvTranspose, :DepthwiseConv, :CrossCor]
+  @eval begin
+    function loadto!(m::$T, m̄::$T)
+      if (size(m.weight) != size(m̄.weight)) || (size(m.bias) != size(m̄.bias))
+        throw(DimensionMismatch("Tried to load $m̄ into $m but the parameter sizes do not match."))
+      else
+        return fmap(loadto!, m, m̄)
+      end
+    end
```

Comment:
I wonder if part of this could be pulled into a helper function. Maybe
```suggestion
    function loadto_helper!(m, m̄, ps)
      for p in ps
        if size(getproperty(m, p)) != size(getproperty(m̄, p))
          throw(DimensionMismatch("Tried to load $m̄ into $m but the parameter sizes do not match."))
        end
      end
      return fmap(loadto!, m, m̄)
    end
```
And then
```suggestion
  loadto!(m::$T, m̄::$T) = loadto_helper!(m, m̄, (:weight, :bias)) 
```